atomics/CMakeLists.txt

```diff
@@ -23,9 +23,37 @@ endif()
 blt_add_library(NAME desul_atomics
                 HEADERS include/desul/atomics.hpp
                 ${SOURCES_ARG}
-                INCLUDES include
                 ${DEPENDS_ARG})
 
-# TODO: Add tests and install targets if needed.
+
+if(ENABLE_CUDA OR ENABLE_HIP)
```

Comment:
Why is it not unconditionally added?
What was wrong with the `INCLUDES` option with `blt_add_library`?

Reply:
Depending on whether there is a source file that will be compiled into the library or not, a different include property must be used. From my reply to Tom:

INTERFACE directories are only used for targets that depend on the target, but not the target itself. Without the PUBLIC include, the source files would fail to find the includes (it has to be written this way since you cannot set PUBLIC include directories on an INTERFACE target) - and desul_atomics is an interface target when built without cuda/hip.

Reply:
Good point! Yes, I think that `INCLUDES` for `blt_add_library` will work as long as it can take the generator expressions.

Reply:
Just took a peek at ``blt_add_library``, definitely seems like a better idea to use that in place of the ``target_include_directories``. Do we need to split up the source and header files into the ``SOURCES`` and ``HEADERS`` flags to achieve what we're doing right now, or does ``INCLUDES`` take care of that conflict?